In [1]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns 
%matplotlib inline 

In [8]:
df = pd.read_csv('DataAnalyst.csv',index_col=0)
df.head()

Job Title  \
0  Data Analyst, Center on Immigration and Justic...   
1                               Quality Data Analyst   
2  Senior Data Analyst, Insights & Analytics Team...   
3                                       Data Analyst   
4                             Reporting Data Analyst   

              Salary Estimate  \
0  $37K-$66K (Glassdoor est.)   
1  $37K-$66K (Glassdoor est.)   
2  $37K-$66K (Glassdoor est.)   
3  $37K-$66K (Glassdoor est.)   
4  $37K-$66K (Glassdoor est.)   

                                     Job Description  Rating  \
0  Are you eager to roll up your sleeves and harn...     3.2   
1  Overview\n\nProvides analytical and technical ...     3.8   
2  We’re looking for a Senior Data Analyst who ha...     3.4   
3  Requisition NumberRR-0001939\nRemote:Yes\nWe c...     4.1   
4  ABOUT FANDUEL GROUP\n\nFanDuel Group is a worl...     3.9   

                              Company Name      Location  Headquarters  \
0           Vera Institute of Justice\n3.2  New York, NY  New York, NY   
1  Visiting Nurse Service of New York\n3.8  New York, NY  New York, NY   
2                         Squarespace\n3.4  New York, NY  New York, NY   
3                            Celerity\n4.1  New York, NY    McLean, VA   
4                             FanDuel\n3.9  New York, NY  New York, NY   

                     Size  Founded               Type of ownership  \
0    201 to 500 employees     1961          Nonprofit Organization   
1        10000+ employees     1893          Nonprofit Organization   
2  1001 to 5000 employees     2003               Company - Private   
3    201 to 500 employees     2002  Subsidiary or Business Segment   
4   501 to 1000 employees     2009               Company - Private   

                           Industry                            Sector  \
0                 Social Assistance                        Non-Profit   
1  Health Care Services & Hospitals                       Health Care   
2                          Internet            Information Technology   
3                       IT Services            Information Technology   
4               Sports & Recreation  Arts, Entertainment & Recreation   

                      Revenue Competitors Easy Apply  
0  $100 to $500 million (USD)          -1       True  
1      $2 to $5 billion (USD)          -1         -1  
2    Unknown / Non-Applicable     GoDaddy         -1  
3   $50 to $100 million (USD)          -1         -1  
4  $100 to $500 million (USD)  DraftKings       True

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2253 entries, 0 to 2252
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         2253 non-null   int64  
 1   Job Title          2253 non-null   object 
 2   Salary Estimate    2253 non-null   object 
 3   Job Description    2253 non-null   object 
 4   Rating             2253 non-null   float64
 5   Company Name       2252 non-null   object 
 6   Location           2253 non-null   object 
 7   Headquarters       2253 non-null   object 
 8   Size               2253 non-null   object 
 9   Founded            2253 non-null   int64  
 10  Type of ownership  2253 non-null   object 
 11  Industry           2253 non-null   object 
 12  Sector             2253 non-null   object 
 13  Revenue            2253 non-null   object 
 14  Competitors        2253 non-null   object 
 15  Easy Apply         2253 non-null   object 
dtypes: float64(1), int64(2),

In [14]:
df.columns

Index(['Job Title', 'Salary Estimate', 'Job Description', 'Rating',
       'Company Name', 'Location', 'Headquarters', 'Size', 'Founded',
       'Type of ownership', 'Industry', 'Sector', 'Revenue', 'Competitors',
       'Easy Apply'],
      dtype='object')

I'D LIKE TO SEE AN EXAMPLE OF A FULL ROW, TO KNOW WHAT I CAN ANALYZE 

In [13]:
df.loc[4]

Job Title                                       Reporting Data Analyst
Salary Estimate                             $37K-$66K (Glassdoor est.)
Job Description      ABOUT FANDUEL GROUP\n\nFanDuel Group is a worl...
Rating                                                             3.9
Company Name                                              FanDuel\n3.9
Location                                                  New York, NY
Headquarters                                              New York, NY
Size                                             501 to 1000 employees
Founded                                                           2009
Type of ownership                                    Company - Private
Industry                                           Sports & Recreation
Sector                                Arts, Entertainment & Recreation
Revenue                                     $100 to $500 million (USD)
Competitors                                                 DraftKings
Easy A

I AM REALLY INTERESTED IN THE COMPANY RATNGS, AND I'D LIKE TO ANALYZE THE RATINGS COLUMN AND FILTER COMPANIES ACCORDING TO THAT

In [15]:
df.Rating.max()

5.0

In [16]:
df.Rating.min()

-1.0

LOL MINIMUM RATING OF -1, INTERESTING 

In [28]:
df[df['Rating'] == 5.0 ]

Job Title  \
26                 Data + Business Intelligence Analyst   
42                                         Data Analyst   
78                                         Data Analyst   
81                                         Data Analyst   
106                                Data/Systems Analyst   
...                                                 ...   
2206                             Marketing Data Analyst   
2207                                       Data Analyst   
2220  MDG Functional Data Analyst-Greenwood Village, CO   
2221                                Data Analyst Junior   
2226                              Data Analyst - Junior   

                  Salary Estimate  \
26     $37K-$66K (Glassdoor est.)   
42     $46K-$87K (Glassdoor est.)   
78     $51K-$88K (Glassdoor est.)   
81     $51K-$88K (Glassdoor est.)   
106    $51K-$87K (Glassdoor est.)   
...                           ...   
2206   $49K-$91K (Glassdoor est.)   
2207   $49K-$91K (Glassdoor est.)   
2220  $57K-$100K (Glassdoor est.)   
2221  $57K-$100K (Glassdoor est.)   
2226  $57K-$100K (Glassdoor est.)   

                                        Job Description  Rating  \
26    We are building Duolingo's newly formed FP&A /...     5.0   
42    Whip Media Group's products, including Mediamo...     5.0   
78    Job Description\nMulberry is disrupting a $40B...     5.0   
81    Los Angeles, CA | New York, NY\n\nEmail career...     5.0   
106   60 months experience in SQL/SQL TOAD;\n60 mont...     5.0   
...                                                 ...     ...   
2206  Gloo, headquartered in beautiful Boulder, Colo...     5.0   
2207  We are looking for data analysts, data wrangle...     5.0   
2220  MDG Functional Data Analyst\nLocation: Greenwo...     5.0   
2221  Job Description\nJob description\nInterpret da...     5.0   
2226  Job Description\nJob description\nInterpret da...     5.0   

                               Company Name                         Location  \
26                            Duolingo\n5.0                     New York, NY   
42                    Whip Media Group\n5.0                     New York, NY   
78                 Mulberry Technology\n5.0                     New York, NY   
81                           EDO, Inc.\n5.0                     New York, NY   
106   Crossfire Consulting Corporation\n5.0                     New York, NY   
...                                     ...                              ...   
2206                Goldstone Partners\n5.0                      Boulder, CO   
2207               xentity corporation\n5.0                       Denver, CO   
2220                   Georgia IT Inc.\n5.0  Greenwood Village, Arapahoe, CO   
2221  Staffigo Technical Services, LLC\n5.0                       Denver, CO   
2226  Staffigo Technical Services, LLC\n5.0                       Denver, CO   

               Headquarters                  Size  Founded  Type of ownership  \
26           Pittsburgh, PA  201 to 500 employees     2011  Company - Private   
42         Santa Monica, CA   51 to 200 employees     2016  Company - Private   
78    New York, NY (US), NY     1 to 50 employees       -1  Company - Private   
81          Los Angeles, CA     1 to 50 employees       -1  Company - Private   
106    Yorktown Heights, NY   51 to 200 employees       -1  Company - Private   
...                     ...                   ...      ...                ...   
2206             Denver, CO     1 to 50 employees       -1  Company - Private   
2207           Lakewood, CO   51 to 200 employees     2001  Company - Private   
2220         Alpharetta, GA     1 to 50 employees       -1  Company - Private   
2221          Woodridge, IL   51 to 200 employees     2008  Company - Private   
2226          Woodridge, IL   51 to 200 employees     2008  Company - Private   

                    Industry                  Sector  \
26                  Internet  Information Technology   
42                        -1             

WE HAVE FILTERED THE COMPANIES WITH A 5.0 RATING 

In [26]:
fs = df[df['Rating'] == 5.0 ] #FS here stands for five-star
fs.head()

Job Title             Salary Estimate  \
26   Data + Business Intelligence Analyst  $37K-$66K (Glassdoor est.)   
42                           Data Analyst  $46K-$87K (Glassdoor est.)   
78                           Data Analyst  $51K-$88K (Glassdoor est.)   
81                           Data Analyst  $51K-$88K (Glassdoor est.)   
106                  Data/Systems Analyst  $51K-$87K (Glassdoor est.)   

                                       Job Description  Rating  \
26   We are building Duolingo's newly formed FP&A /...     5.0   
42   Whip Media Group's products, including Mediamo...     5.0   
78   Job Description\nMulberry is disrupting a $40B...     5.0   
81   Los Angeles, CA | New York, NY\n\nEmail career...     5.0   
106  60 months experience in SQL/SQL TOAD;\n60 mont...     5.0   

                              Company Name      Location  \
26                           Duolingo\n5.0  New York, NY   
42                   Whip Media Group\n5.0  New York, NY   
78                Mulberry Technology\n5.0  New York, NY   
81                          EDO, Inc.\n5.0  New York, NY   
106  Crossfire Consulting Corporation\n5.0  New York, NY   

              Headquarters                  Size  Founded  Type of ownership  \
26          Pittsburgh, PA  201 to 500 employees     2011  Company - Private   
42        Santa Monica, CA   51 to 200 employees     2016  Company - Private   
78   New York, NY (US), NY     1 to 50 employees       -1  Company - Private   
81         Los Angeles, CA     1 to 50 employees       -1  Company - Private   
106   Yorktown Heights, NY   51 to 200 employees       -1  Company - Private   

       Industry                  Sector                   Revenue Competitors  \
26     Internet  Information Technology  $10 to $25 million (USD)          -1   
42           -1                      -1  Unknown / Non-Applicable          -1   
78           -1                      -1  Unknown / Non-Applicable          -1   
81     Internet  Information Technology  Unknown / Non-Applicable          -1   
106  Consulting       Business Services   $5 to $10 million (USD)          -1   

    Easy Apply  
26          -1  
42          -1  
78          -1  
81          -1  
106         -1

In [27]:
fs.index

Int64Index([  26,   42,   78,   81,  106,  114,  124,  146,  206,  211,
            ...
            2167, 2173, 2179, 2181, 2191, 2206, 2207, 2220, 2221, 2226],
           dtype='int64', length=131)

In [25]:
fs['Company Name'].nunique()

61

ALTHOUGH WE HAVE 131 ROWS, WE HAVE JUST 61 DIFFERENT COMPANIES 

In [31]:
fs['Company Name'].value_counts()

Staffigo Technical Services, LLC\n5.0    58
Georgia IT Inc.\n5.0                      4
WB Technology\n5.0                        3
Jobot\n5.0                                2
Professional Resource Group\n5.0          2
                                         ..
Connect International\n5.0                1
Quinn Group\n5.0                          1
Priceonomics\n5.0                         1
PriceSenz\n5.0                            1
Resiliency LLC\n5.0                       1
Name: Company Name, Length: 61, dtype: int64

In [34]:
fs.head()

Job Title             Salary Estimate  \
26   Data + Business Intelligence Analyst  $37K-$66K (Glassdoor est.)   
42                           Data Analyst  $46K-$87K (Glassdoor est.)   
78                           Data Analyst  $51K-$88K (Glassdoor est.)   
81                           Data Analyst  $51K-$88K (Glassdoor est.)   
106                  Data/Systems Analyst  $51K-$87K (Glassdoor est.)   

                                       Job Description  Rating  \
26   We are building Duolingo's newly formed FP&A /...     5.0   
42   Whip Media Group's products, including Mediamo...     5.0   
78   Job Description\nMulberry is disrupting a $40B...     5.0   
81   Los Angeles, CA | New York, NY\n\nEmail career...     5.0   
106  60 months experience in SQL/SQL TOAD;\n60 mont...     5.0   

                              Company Name      Location  \
26                           Duolingo\n5.0  New York, NY   
42                   Whip Media Group\n5.0  New York, NY   
78                Mulberry Technology\n5.0  New York, NY   
81                          EDO, Inc.\n5.0  New York, NY   
106  Crossfire Consulting Corporation\n5.0  New York, NY   

              Headquarters                  Size  Founded  Type of ownership  \
26          Pittsburgh, PA  201 to 500 employees     2011  Company - Private   
42        Santa Monica, CA   51 to 200 employees     2016  Company - Private   
78   New York, NY (US), NY     1 to 50 employees       -1  Company - Private   
81         Los Angeles, CA     1 to 50 employees       -1  Company - Private   
106   Yorktown Heights, NY   51 to 200 employees       -1  Company - Private   

       Industry                  Sector                   Revenue Competitors  \
26     Internet  Information Technology  $10 to $25 million (USD)          -1   
42           -1                      -1  Unknown / Non-Applicable          -1   
78           -1                      -1  Unknown / Non-Applicable          -1   
81     Internet  Information Technology  Unknown / Non-Applicable          -1   
106  Consulting       Business Services   $5 to $10 million (USD)          -1   

    Easy Apply  
26          -1  
42          -1  
78          -1  
81          -1  
106         -1

In [49]:
fs.head()

Job Title             Salary Estimate  \
26   Data + Business Intelligence Analyst  $37K-$66K (Glassdoor est.)   
42                           Data Analyst  $46K-$87K (Glassdoor est.)   
78                           Data Analyst  $51K-$88K (Glassdoor est.)   
81                           Data Analyst  $51K-$88K (Glassdoor est.)   
106                  Data/Systems Analyst  $51K-$87K (Glassdoor est.)   

                                       Job Description  Rating  \
26   We are building Duolingo's newly formed FP&A /...     5.0   
42   Whip Media Group's products, including Mediamo...     5.0   
78   Job Description\nMulberry is disrupting a $40B...     5.0   
81   Los Angeles, CA | New York, NY\n\nEmail career...     5.0   
106  60 months experience in SQL/SQL TOAD;\n60 mont...     5.0   

                              Company Name      Location  \
26                           Duolingo\n5.0  New York, NY   
42                   Whip Media Group\n5.0  New York, NY   
78                Mulberry Technology\n5.0  New York, NY   
81                          EDO, Inc.\n5.0  New York, NY   
106  Crossfire Consulting Corporation\n5.0  New York, NY   

              Headquarters                  Size  Founded  Type of ownership  \
26          Pittsburgh, PA  201 to 500 employees     2011  Company - Private   
42        Santa Monica, CA   51 to 200 employees     2016  Company - Private   
78   New York, NY (US), NY     1 to 50 employees       -1  Company - Private   
81         Los Angeles, CA     1 to 50 employees       -1  Company - Private   
106   Yorktown Heights, NY   51 to 200 employees       -1  Company - Private   

       Industry                  Sector                   Revenue Competitors  \
26     Internet  Information Technology  $10 to $25 million (USD)          -1   
42           -1                      -1  Unknown / Non-Applicable          -1   
78           -1                      -1  Unknown / Non-Applicable          -1   
81     Internet  Information Technology  Unknown / Non-Applicable          -1   
106  Consulting       Business Services   $5 to $10 million (USD)          -1   

    Easy Apply               Company   
26          -1                  Duoli  
42          -1  Whip Media Group\n5.0  
78          -1          Mulberry Tech  
81          -1                 EDO, I  
106         -1           Crossfire Co

In [62]:
fs.head()

Job Title             Salary Estimate  \
26   Data + Business Intelligence Analyst  $37K-$66K (Glassdoor est.)   
42                           Data Analyst  $46K-$87K (Glassdoor est.)   
78                           Data Analyst  $51K-$88K (Glassdoor est.)   
81                           Data Analyst  $51K-$88K (Glassdoor est.)   
106                  Data/Systems Analyst  $51K-$87K (Glassdoor est.)   

                                       Job Description  Rating  \
26   We are building Duolingo's newly formed FP&A /...     5.0   
42   Whip Media Group's products, including Mediamo...     5.0   
78   Job Description\nMulberry is disrupting a $40B...     5.0   
81   Los Angeles, CA | New York, NY\n\nEmail career...     5.0   
106  60 months experience in SQL/SQL TOAD;\n60 mont...     5.0   

                         Company Name      Location           Headquarters  \
26                           Duolingo  New York, NY         Pittsburgh, PA   
42                   Whip Media Group  New York, NY       Santa Monica, CA   
78                Mulberry Technology  New York, NY  New York, NY (US), NY   
81                          EDO, Inc.  New York, NY        Los Angeles, CA   
106  Crossfire Consulting Corporation  New York, NY   Yorktown Heights, NY   

                     Size  Founded  Type of ownership    Industry  \
26   201 to 500 employees     2011  Company - Private    Internet   
42    51 to 200 employees     2016  Company - Private          -1   
78      1 to 50 employees       -1  Company - Private          -1   
81      1 to 50 employees       -1  Company - Private    Internet   
106   51 to 200 employees       -1  Company - Private  Consulting   

                     Sector                   Revenue Competitors Easy Apply  
26   Information Technology  $10 to $25 million (USD)          -1         -1  
42                       -1  Unknown / Non-Applicable          -1         -1  
78                       -1  Unknown / Non-Applicable          -1         -1  
81   Information Technology  Unknown / Non-Applicable          -1         -1  
106       Business Services   $5 to $10 million (USD)          -1         -1

I didn't like the '\n' on the company names might've been a typo for new line indicating the rating but 

In [60]:
fs['Company Name'] = fs['Company Name'].apply(lambda x : x.split("\n")[0])

<ipython-input-60-686fc389c0bf>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  fs['Company Name'] = fs['Company Name'].apply(lambda x : x.split("\n")[0])


In [61]:
fs.head()

Job Title             Salary Estimate  \
26   Data + Business Intelligence Analyst  $37K-$66K (Glassdoor est.)   
42                           Data Analyst  $46K-$87K (Glassdoor est.)   
78                           Data Analyst  $51K-$88K (Glassdoor est.)   
81                           Data Analyst  $51K-$88K (Glassdoor est.)   
106                  Data/Systems Analyst  $51K-$87K (Glassdoor est.)   

                                       Job Description  Rating  \
26   We are building Duolingo's newly formed FP&A /...     5.0   
42   Whip Media Group's products, including Mediamo...     5.0   
78   Job Description\nMulberry is disrupting a $40B...     5.0   
81   Los Angeles, CA | New York, NY\n\nEmail career...     5.0   
106  60 months experience in SQL/SQL TOAD;\n60 mont...     5.0   

                         Company Name      Location           Headquarters  \
26                           Duolingo  New York, NY         Pittsburgh, PA   
42                   Whip Media Group  New York, NY       Santa Monica, CA   
78                Mulberry Technology  New York, NY  New York, NY (US), NY   
81                          EDO, Inc.  New York, NY        Los Angeles, CA   
106  Crossfire Consulting Corporation  New York, NY   Yorktown Heights, NY   

                     Size  Founded  Type of ownership    Industry  \
26   201 to 500 employees     2011  Company - Private    Internet   
42    51 to 200 employees     2016  Company - Private          -1   
78      1 to 50 employees       -1  Company - Private          -1   
81      1 to 50 employees       -1  Company - Private    Internet   
106   51 to 200 employees       -1  Company - Private  Consulting   

                     Sector                   Revenue Competitors Easy Apply  
26   Information Technology  $10 to $25 million (USD)          -1         -1  
42                       -1  Unknown / Non-Applicable          -1         -1  
78                       -1  Unknown / Non-Applicable          -1         -1  
81   Information Technology  Unknown / Non-Applicable          -1         -1  
106       Business Services   $5 to $10 million (USD)          -1         -1

Now I am wondering what cities would be best for Data Analysts or anyone involved in Data Science in general

In [63]:
fs['Location_State'] = fs['Location'].apply(lambda x: x.split(',')[1)

<ipython-input-63-f8c32ad515bb>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  fs['Location_State'] = fs['Location'].apply(lambda x: x.split(',')[1])


In [64]:
fs.Location_State

26             NY
42             NY
78             NY
81             NY
106            NY
          ...    
2206           CO
2207           CO
2220     Arapahoe
2221           CO
2226           CO
Name: Location_State, Length: 131, dtype: object

Checking for the common states 

In [66]:
fs.Location_State.value_counts()

 CA          38
 NY          19
 TX          14
 FL           9
 CO           9
 IL           8
 NC           7
 VA           5
 OH           4
 PA           4
 WA           3
 IN           3
 GA           3
 AZ           2
 Arapahoe     1
 DE           1
 UT           1
Name: Location_State, dtype: int64

NOW THAT WE HAVE SEEN THE MOST COMMON STATES, I'D LIKE TO COMPARE SALARIES 

In [73]:
fs.dtypes

Job Title             object
Salary Estimate       object
Job Description       object
Rating               float64
Company Name          object
Location              object
Headquarters          object
Size                  object
Founded                int64
Type of ownership     object
Industry              object
Sector                object
Revenue               object
Competitors           object
Easy Apply            object
Location_State        object
dtype: object

In [96]:
fs[fs['Location_State'] == ' CA']['Salary Estimate'].iloc[3].split()[0].split('-')[0][1:-1]

'55'

In [106]:
fs['low pay'] = fs['Salary Estimate'].apply(lambda x: x.split()[0].split('-')[0][1:-1])

<ipython-input-106-730cb5f3db8a>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  fs['low pay'] = fs['Salary Estimate'].apply(lambda x: x.split()[0].split('-')[0][1:-1])


In [107]:
fs['high pay'] = fs['Salary Estimate'].apply(lambda x: x.split()[0].split('-')[1][1:-1])

<ipython-input-107-897a39480204>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  fs['high pay'] = fs['Salary Estimate'].apply(lambda x: x.split()[0].split('-')[1][1:-1])


In [108]:
fs.head()

Job Title             Salary Estimate  \
26   Data + Business Intelligence Analyst  $37K-$66K (Glassdoor est.)   
42                           Data Analyst  $46K-$87K (Glassdoor est.)   
78                           Data Analyst  $51K-$88K (Glassdoor est.)   
81                           Data Analyst  $51K-$88K (Glassdoor est.)   
106                  Data/Systems Analyst  $51K-$87K (Glassdoor est.)   

                                       Job Description  Rating  \
26   We are building Duolingo's newly formed FP&A /...     5.0   
42   Whip Media Group's products, including Mediamo...     5.0   
78   Job Description\nMulberry is disrupting a $40B...     5.0   
81   Los Angeles, CA | New York, NY\n\nEmail career...     5.0   
106  60 months experience in SQL/SQL TOAD;\n60 mont...     5.0   

                         Company Name      Location           Headquarters  \
26                           Duolingo  New York, NY         Pittsburgh, PA   
42                   Whip Media Group  New York, NY       Santa Monica, CA   
78                Mulberry Technology  New York, NY  New York, NY (US), NY   
81                          EDO, Inc.  New York, NY        Los Angeles, CA   
106  Crossfire Consulting Corporation  New York, NY   Yorktown Heights, NY   

                     Size  Founded  Type of ownership    Industry  \
26   201 to 500 employees     2011  Company - Private    Internet   
42    51 to 200 employees     2016  Company - Private          -1   
78      1 to 50 employees       -1  Company - Private          -1   
81      1 to 50 employees       -1  Company - Private    Internet   
106   51 to 200 employees       -1  Company - Private  Consulting   

                     Sector                   Revenue Competitors Easy Apply  \
26   Information Technology  $10 to $25 million (USD)          -1         -1   
42                       -1  Unknown / Non-Applicable          -1         -1   
78                       -1  Unknown / Non-Applicable          -1         -1   
81   Information Technology  Unknown / Non-Applicable          -1         -1   
106       Business Services   $5 to $10 million (USD)          -1         -1   

    Location_State low pay high pay  
26              NY      37       66  
42              NY      46       87  
78              NY      51       88  
81              NY      51       88  
106             NY      51       87

In [110]:
fs = fs.astype({"low pay": int, "high pay": int})

In [112]:
fs['Average Pay'] = (fs['low pay'] + fs['high pay'])/2

TIME TO CHECK FIVE STAR COMPANIES WITH THE HIGHEST AND LOWEST PAY 

In [119]:
fs.nlargest(5,'Average Pay')

Job Title               Salary Estimate  \
1484              AI Insights Data Analyst  $110K-$190K (Glassdoor est.)   
1919  Data Scientist/Analyst - Wilbur Labs   $99K-$178K (Glassdoor est.)   
1929                   Data Studio Analyst   $99K-$178K (Glassdoor est.)   
1944               Operations Data Analyst   $93K-$159K (Glassdoor est.)   
1957                          Data Analyst   $93K-$159K (Glassdoor est.)   

                                        Job Description  Rating  Company Name  \
1484  As an AI Insights Data Analyst at Moveworks, y...     5.0     Moveworks   
1919  About Wilbur Labs\nWilbur Labs is a startup st...     5.0   Wilbur Labs   
1929  At Priceonomics, our core skill is making data...     5.0  Priceonomics   
1944  About Turn/River\n\nTurn/River Capital is a le...     5.0          Turn   
1957  5+ years of Experience in SQL development meth...     5.0   Softova Inc   

               Location       Headquarters                 Size  Founded  \
1484  Mountain View, CA  Mountain View, CA  51 to 200 employees     2016   
1919  San Francisco, CA  San Francisco, CA    1 to 50 employees     2016   
1929  San Francisco, CA  San Francisco, CA    1 to 50 employees       -1   
1944  San Francisco, CA  San Francisco, CA    1 to 50 employees       -1   
1957      San Mateo, CA     Piscataway, NJ    1 to 50 employees       -1   

      Type of ownership                                 Industry  \
1484  Company - Private  Enterprise Software & Network Solutions   
1919  Company - Private                                 Internet   
1929  Company - Private                                 Internet   
1944  Company - Private         Venture Capital & Private Equity   
1957  Company - Private                              IT Services   

                      Sector                     Revenue Competitors  \
1484  Information Technology    Unknown / Non-Applicable          -1   
1919  Information Technology    Unknown / Non-Applicable          -1   
1929  Information Technology  Less than $1 million (USD)          -1   
1944                 Finance    Unknown / Non-Applicable          -1   
1957  Information Technology      $1 to $5 million (USD)          -1   

     Easy Apply Location_State  low pay  high pay  Average Pay  
1484       True             CA      110       190        150.0  
1919         -1             CA       99       178        138.5  
1929         -1             CA       99       178        138.5  
1944         -1             CA       93       159        126.0  
1957         -1             CA       93       159        126.0

In [120]:
fs.nsmallest(5,'Average Pay')

Job Title             Salary Estimate  \
1322    Game Data Analyst  $24K-$48K (Glassdoor est.)   
1339  Junior Data Analyst  $24K-$48K (Glassdoor est.)   
446   Senior Data Analyst  $26K-$47K (Glassdoor est.)   
1706  Data Analyst Junior  $35K-$42K (Glassdoor est.)   
600   Data Analyst Junior  $35K-$45K (Glassdoor est.)   

                                        Job Description  Rating  \
1322  About PeopleFun\n\nPeopleFun is a rapidly grow...     5.0   
1339  Job Description\nJob description\nInterpret da...     5.0   
446   Job Description\nReq Notes:\nThis is a data en...     5.0   
1706  Job Description\nJob description\nInterpret da...     5.0   
600   Job Description\nJob description\nInterpret da...     5.0   

                          Company Name            Location    Headquarters  \
1322                         PeopleFun      Richardson, TX  Richardson, TX   
1339  Staffigo Technical Services, LLC          Dallas, TX   Woodridge, IL   
446                     Resiliency LLC  Salt Lake City, UT    Stamford, CT   
1706  Staffigo Technical Services, LLC          Austin, TX   Woodridge, IL   
600   Staffigo Technical Services, LLC     Los Angeles, CA   Woodridge, IL   

                     Size  Founded  Type of ownership     Industry  \
1322    1 to 50 employees       -1  Company - Private  Audiovisual   
1339  51 to 200 employees     2008  Company - Private  IT Services   
446     1 to 50 employees       -1  Company - Private           -1   
1706  51 to 200 employees     2008  Company - Private  IT Services   
600   51 to 200 employees     2008  Company - Private  IT Services   

                                Sector                    Revenue Competitors  \
1322  Arts, Entertainment & Recreation   Unknown / Non-Applicable          -1   
1339            Information Technology  $50 to $100 million (USD)          -1   
446                                 -1   Unknown / Non-Applicable          -1   
1706            Information Technology  $50 to $100 million (USD)          -1   
600             Information Technology  $50 to $100 million (USD)          -1   

     Easy Apply Location_State  low pay  high pay  Average Pay  
1322         -1             TX       24        48         36.0  
1339         -1             TX       24        48         36.0  
446          -1             UT       26        47         36.5  
1706         -1             TX       35        42         38.5  
600          -1             CA       35        45         40.0

WE CAN TELL THAT THE BEST STATES TO MOVE TO AS A DATA ANAL